pkg/bootstrap/confirm/tasks.go

```diff
@@ -236,8 +248,22 @@ Warning:
 	}
 	fmt.Println()
 
-	reader := bufio.NewReader(os.Stdin)
+	if k8sVersion, err := versionutil.ParseGeneric(u.KubeConf.Cluster.Kubernetes.Version); err == nil {
+		if cri, ok := u.PipelineCache.GetMustString(common.ClusterNodeCRIRuntimes); ok {
+			k8sV124 := versionutil.MustParseSemantic("v1.24.0")
+			if k8sVersion.AtLeast(k8sV124) && versionutil.MustParseSemantic(currentK8sVersion).LessThan(k8sV124) && strings.Contains(cri, "docker") {
+				fmt.Println("[Notice]")
+				fmt.Println("Dockershim has been removed from the Kubernetes project as of release 1.24.")
```

Comment:
Kubernetes v1.24 and later no longer support dockershim.

Reply:
Done.